backend/tracim_backend/lib/utils/app.py

```diff
@@ -0,0 +1,110 @@
+from abc import ABC
+import typing
+
+from hapic.ext.pyramid import PyramidContext
+from pyramid.config import Configurator
+
+from tracim_backend.app_models.applications import TracimApplicationInContext
+from tracim_backend.models.roles import WorkspaceRoles
+
+if typing.TYPE_CHECKING:
+    from tracim_backend.config import CFG  # noqa:F401
+    from tracim_backend.app_models.contents import ContentStatus  # noqa:F401
+
+
+class TracimContentType(ABC):
+    def __init__(
+        self,
+        slug: str,
+        fa_icon: str,
+        label: str,
+        creation_label: str,
+        available_statuses: typing.List["ContentStatus"],
+        app: "TracimApplication" = None,
+        slug_alias: typing.List[str] = None,
+        allow_sub_content: bool = False,
+        file_extension: typing.Optional[str] = None,
+        minimal_role_content_creation: WorkspaceRoles = WorkspaceRoles.CONTRIBUTOR,
+    ) -> None:
+        self.slug = slug
+        self.fa_icon = fa_icon
+        self.label = label
+        self.creation_label = creation_label
+        self.available_statuses = available_statuses
+        self.slug_alias = slug_alias
+        self.allow_sub_content = allow_sub_content
+        self.file_extension = file_extension
+        self.minimal_role_content_creation = minimal_role_content_creation
+        self.app = app
+
+
+class TracimApplication(ABC):
```

Comment:
TracimApplication Spec.

Reply:
As this class is a spec (specification?), you should explain in docstring how to use / inherit / reimplement it.

Reply:
done.